### PR TITLE
Multi tenancy

### PR DESCRIPTION
**Pull request by galthaus**
Add new controls for new multi-tenancy objects.
Also add tenant_id to the objects that now have ownership by tenant.

### CODE REVIEW

File: datatypes/tenant.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package datatypes
+
+type Tenant struct {
+	SimpleID
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	ParentID    int64  `json:"deployment_id"`
```
Comment (VictorLowther) [inline datatypes/tenant.go]: parent_id, surely?

File: datatypes/user.go [hunk 2]
```diff
@@ -2,10 +2,12 @@ package datatypes
...
 	SimpleID
-	Name   string `json:"username"`
-	Email  string `json:"email"`
-	Admin  bool   `json:"is_admin"`
-	Locked bool   `json:"locked"`
+	Name            string `json:"username"`
+	Email           string `json:"email"`
+	Admin           bool   `json:"is_admin"`
+	TenantID        int64  `json:"tenant_id"`
+	CurrentTenantID int64  `json:"current_tenant_id"`
```
Comment (VictorLowther) [inline datatypes/user.go]: What does this wind up being for?

Comment (galthaus) [inline datatypes/user.go]: tenant_id is who owns you.
Current_tenant_id is the tenant you are operating in.

File: datatypes/tenant.go (hunk 1 above)
Comment (galthaus) [inline datatypes/tenant.go]: oops

**Comment by VictorLowther** [comment]
LGTM.  Want to pull it now, or wait until the rest of the code is reviewed and pulled?

**Comment by galthaus** [comment]
wait - may need to tweak api for other changes.

**Pull request merged**